### PR TITLE
PR for filter pill changes

### PR DESCRIPTION
**Pull request by yerramshilpa**
added a update items method in the selected result set

### CODE REVIEW

File: src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts [hunk 1]
```diff
@@ -85,6 +85,19 @@ export class HierarchicalTreeSelectedItemModel {
...
     }
+    /**
+     * updates an array of items to the list and will not add duplicate items
+     * keyfield is used to determine uniqueness of the item added
+     * @param items 
+     * @param keyField 
+     */
+    updateItems(items: object[], keyField: string) {
+        this.items =[];
+        console.log(items,'selected items')
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts]: I need some method to update the existing record.

File: src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts [hunk 2]
```diff
@@ -85,6 +85,19 @@ export class HierarchicalTreeSelectedItemModel {
...
+        for (let i = 0; i < items.length; i++) {
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts]: That adds to the list I need something to update.

File: src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts]: I will remove

File: src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomastighe) [reply src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts]: This method already does what you are doing.
    addItems(items: object[], keyField: string) {
        for (let i = 0; i < items.length; i++) {
            this.addItem(items[i], keyField);
        }
}

File: src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (thomastighe) [reply src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts]: If you are updating the items in the list you could find the item which there is already a method for and just replace the item at the indices

**Pull request merged**